src/preferences/colorpaletteeditor.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <QComboBox>
+#include <QPushButton>
+#include <QTableView>
+#include <QWidget>
+
+#include "preferences/colorpaletteeditormodel.h"
+#include "preferences/usersettings.h"
+#include "util/parented_ptr.h"
+
+// Widget for viewing, adding, editing and removing color palettes that can be
+// used for track/hotcue colors.
+class ColorPaletteEditor : public QWidget {
+    Q_OBJECT
+  public:
+    ColorPaletteEditor(QWidget* parent = nullptr);
+    void setConfig(UserSettingsPointer pConfig) {
```

Comment:
- Why not configure() or initialize()? This functions is supposed to do more than just setting a member.
- Maybe even add DEBUG_ASSERT(!m_pConfig) if it is only supposed to be called once.
- Better hide the actual implementation in the .cpp file

Reply:
OK. That function is just a crutch because I don't know how to pass custom args to the constructor of Qt Designer widgets.